### ISSUE TITLE
Error: missing attributes in ROM::Relation

### ISSUE THREAD

**Issue by greggilbert** [hanami/model#416]
Hey. I'm getting this error as part of my model: `Hanami::Model::Error: missing attributes in ROM::Relation::Name(commits) schema: :payload`. I think I'm doing everything right, but maybe not?

```
# migration
Hanami::Model.migration do
  change do
    create_table :commits do
      # other stuff
      column :payload, :longtext, null: false
    end
  end
end
```

```
# model
class Commit < Hanami::Entity
  attributes do
    attribute :payload,         Types::String
  end
end
```

The error backtraces to hanami/model/config line 134. I just tried deleting the `payload` field from the table, and it works again.

Is there something actually wrong here, or did I make a mistake?

**Update:** I added a proof of concept of this at https://github.com/greggilbert/hanami-model-416.

**Comment by kaikuchn** [comment]
As discussed in the Chat on [July 24, 2017 10:05 AM](https://gitter.im/hanami/chat?at=5975aa52bc464729741c9cd4), the probable cause is that rom-sql does not support mysql types.

**Comment by jodosha** [comment]
@greggilbert @kaikuchn I debugged the problem, this `rom-sql` [method](https://github.com/rom-rb/rom-sql/blob/release-1.0/lib/rom/sql/schema/inferrer.rb#L84) is responsible to build a ROM type for each attribute, but it fails because Sequel doesn't return a "ruby type" for MySQL's `:longtext`.

```ruby
(byebug) type
nil
(byebug) db_type
"longtext"
(byebug) rest
{:default=>nil, :ruby_default=>nil}
```

Because of that lacking information, ROM skips that column/attribute.

---

@flash-gordon @solnic Does it makes sense to have a fallback to string instead to ignore the column?

**Comment by solnic** [comment]
I'd say we should raise an error, as in, we should not infer an attribute when ruby type is nil, then you'll get an 'attribute missing error' and you'll know you need to define it manually.

**Comment by flash-gordon** [comment]
I changed the error message in 4.0 btw https://github.com/rom-rb/rom/blob/7ed4ad8b030210692bc850ae8fbf3e293922b9e1/core/lib/rom/schema/inferrer.rb#L12-L19

**Comment by HeitorMC** [comment]
Hey guys, I ran the PoC that @greggilbert provided, but changed the ruby version to 2.7.7 and this error persists. Also, I checked the [gemspec](https://github.com/hanami/model/blob/main/hanami-model.gemspec#L24) of `hanami-model` and the ROM dependency points to `"~> 3.3", ">= 3.3.3"`, so the fix that @flash-gordon made won't be applied. 

So I was wondering if this issue/bug is still valid, and if I could work on it

**Comment by HeitorMC** [comment]
@kaikuchn @jodosha Hey, it's been a while. I want to check with you if this is an _active_ bug, and if so, can we bump the dependency of `hanami-model` to use the newer version with the error message, available in 4.0? Or, this issue is solved according to the [chat](https://gitter.im/hanami/chat?at=5975aa52bc464729741c9cd4), and can we close it?

**Comment by kaikuchn** [comment]
I just helped @greggilbert to debug the issue. Bumping the dependency of Hanami-Model to ROM 4.0 would be nice but I don't think it's feasible without breaking changes to Hanami-Model.

Since this error has a working solution my suggestion would be to close this issue!
